### ISSUE TITLE
Extra tanh?

### ISSUE THREAD

**Issue by dhzyingz** [pangolulu/neural-network-from-scratch#3]
In the backpropagation part, the first line code writes:

dtanh = softmaxOutput.diff(forward[len(forward)-1][2], y)

So it is activated then sent to softmax? 
I guess for the last layer there is no need to add tanh before softmax? So the code would be:

dtanh = softmaxOutput.diff(forward[len(forward)-1][1], y)
